### ISSUE TITLE
"Safari can't establish a secure connection."

### ISSUE THREAD

**Issue by WolfgangDpunkt** [AdguardTeam/AdGuardHome#5941]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not [ask a question or ask for help](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a)

- [X] I have set up AdGuard Home correctly and [configured clients to use it](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients). (Use the [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) for help with installing and configuring clients.)

### Platform (OS and CPU architecture)

Linux/ARM64

### Installation

Docker

### Setup

On one machine

### AdGuard Home version

v0.107.32

### Action

I tried to open the page https://archive.is/ in Safari on my iPhone.

### Expected result

https://archive.is
<img width="200" alt="grafik" src="https://github.com/AdguardTeam/AdGuardHome/assets/35533844/ab6eafeb-33bb-459f-84ec-51bcad82b92a">

### Actual result

The page has not been available. Error message:
Safari can't establish a secure connection.
<img width="200" alt="grafik" src="https://github.com/AdguardTeam/AdGuardHome/assets/35533844/1287e386-77d3-4d91-8a31-c10689c6bc50">

### Additional information and/or screenshots

With the AdGuard Home filtering disabled, the site can be accessed on the same iPhone without any problems. The site also works without any problems on other devices that are not protected with AdGuard Home.
In the filter log, the page is displayed as unblocked, and other blocking messages are also not displayed in this context. iOS16.5.1
![grafik](https://github.com/AdguardTeam/AdGuardHome/assets/35533844/6c1375d6-a034-492d-840a-4abea4a85e7e)

**Comment by agneevX** [comment]
> Safari can't establish a secure connection.

This error is not related to DNS.

**Comment by WolfgangDpunkt** [comment]
> This error is not related to DNS. 
Thanks for your response. It's quite possible that this is not directly affect by DNS.
But Adguard Home apparently does seem to play a role, since disabling it fixes the error immediately.

There have also been similar observations:
https://forum.adguard.com/index.php?threads/how-to-fix-https-issues.6158/

**Comment by agneevX** [comment]
That issue is simple out of scope of AdGuard Home. 

What AGH does is it forwards DNS queries made by your device to a provider and returns the resulting response back to your device. 

That's it. It does not read or manipulate network traffic or play a role in SNI blocking as apparent from your error.

**Comment by WolfgangDpunkt** [comment]
Thanks for the explanation. I don't doubt that. But apparently what AGH does still sometimes has to do with HTTPS traffic errors. There is even a separate listing for that in the documentation, after all:
https://adguard.com/kb/de/adguard-for-windows/solving-problems/connection-not-trusted/

I'm not sure how comparable it is, but I think AGH may well be a cause _sometimes_ in SSL/HTTPS problems. The basic possibility apparently exists, I just want to emphasize that. :-)

It would be interesting from both a technical and user perspective what is happening here.

**Comment by ainar-g** [comment]
As agneevX mentioned, DNS filtering usually shouldn't cause anything like this, especially if the query isn't filtered.  Some data that could shed light are:

1.  Is the issue only with this site?  Or all sites?  Or some but not others?

2.  What are the actual responses?  Do they contain valid IP addresses?  (You can check those in the query log.)

3.  You've mentioned that disabling the filtering fixes the issue.  How are the responses/IP addresses different?  And while we're at it, which filtering features (parental control, safe browsing, etc.) and filtering lists are you using?

**Comment by WolfgangDpunkt** [comment]
After testing for a while, I have now found the source of the error: I had set Cloudflare DNS (1.1.1.1) as an upstream DNS server in Adguard Home. The target site https://archive.is/ is protected by Cloudflare. Apparently, this two things did not work well together. Since I started using other upstream DNS servers in Adguard Home, everything has been working as expected. I'm not sure why exactly, but that's how the problem was solved.

**Comment by agneevX** [comment]
Ah yes, I recall reading this on HN some time back: https://news.ycombinator.com/item?id=28495204.
